attributes/sdk.rb

```diff
@@ -91,6 +91,8 @@
 default['cdap']['sdk']['install_path'] = '/opt/cdap'
 default['cdap']['sdk']['user'] = 'cdap'
 default['cdap']['sdk']['manage_user'] = true
+default['cdap']['sdk']['profile_d']['node_env'] = 'production'
```

Comment:
You're correct. It's unset for SDK. The SDK is launched via `UserInterfaceService` class, which writes out cConf and sConf to JSON files, then gives the file paths to the `node` process at runtime: `node /opt/cdap/sdk-4.1.1/ui/server.js cConf="/opt/cdap/sdk-4.1.1/data/conf/cdap-config.json" sConf="/opt/cdap/sdk-4.1.1/data/conf/cdap-security-config.json" which is only performed if `NODE_ENV` is anything *except* "production"... `server/config/parser.js`

Now, I've verified that setting this to an empty string *will* write out something to the profile.d file, but none of the UI code cares about anything other than specific options:
- production
- development
- browser-testing

Anything else will look for cConf/sConf on the command line and fall back to the `development` configuration if it isn't found. Setting this to an empty string.